### PR TITLE
Drone S3 Region Support

### PR DESCRIPTION
**Pull request by captn3m0**
Might need changes before merging back.

### CODE REVIEW

File: storage/s3/s3.go [hunk 1]
```diff
@@ -45,13 +44,14 @@ type s3Storage struct {
...
-	}
-	client, err := minio.NewWithCredentials(opts.Endpoint, creds, opts.UseSSL, "")
+	// var creds *credentials.Credentials
+	// if len(opts.Access) != 0 && len(opts.Secret) != 0 {
+	// 	creds = credentials.NewStaticV4(opts.Access, opts.Secret, "")
+	// } else {
+	// 	creds = credentials.NewIAM("")
+	// }
+	// client, err := minio.NewWithCredentials(opts.Endpoint, creds, opts.UseSSL, opts.Region)
+	client, err := minio.NewWithRegion(opts.Endpoint, opts.Access, opts.Secret, opts.UseSSL, opts.Region)
```
Comment (tboerger) [inline storage/s3/s3.go]: This won't get merged like it is, it drops the iam role stuff.

Comment (donny-dont) [reply storage/s3/s3.go]: https://github.com/minio/minio-go/blob/3007bfa46ebc036644e727bb1e0cf3696293de50/api.go#L166-L171

All you have to do is add the opts.Region to the end.

Comment (donny-dont) [reply storage/s3/s3.go]: The main.go changes are fine and NewWithCredentials has a parameter for region. Honestly the part commented out just needs to be reverted and the empty string parameter replaced by the region.

Happy to merge it @Hashfyre with the changes in plugin.go

Comment (donny-dont) [reply storage/s3/s3.go]: The main.go changes are fine and NewWithCredentials has a parameter for region. Honestly the part commented out just needs to be reverted and the empty string parameter replaced by the region.

Happy to merge it @Hashfyre with the changes in plugin.go

**Comment by donny-dont** [comment]
@Hashfyre @captn3m0 will one of you be addressing the feedback here?

**Pull request closed without merging**